rbx_binary/src/deserializer.rs

```diff
@@ -855,7 +855,40 @@ impl<R: Read> BinaryDeserializer<R> {
             Type::ColorSequence => {}
             Type::NumberRange => {}
             Type::Rect => {}
-            Type::PhysicalProperties => {}
+            Type::PhysicalProperties => match canonical_type {
+                VariantType::PhysicalProperties => {
+                    let mut values = Vec::with_capacity(type_info.referents.len());
+
+                    for _ in 0..type_info.referents.len() {
+                        if chunk.read_u8()? == 1 {
+                            values.push(Variant::PhysicalProperties(PhysicalProperties::Custom(
+                                CustomPhysicalProperties {
+                                    density: chunk.read_le_f32()?,
+                                    friction: chunk.read_le_f32()?,
+                                    elasticity: chunk.read_le_f32()?,
+                                    friction_weight: chunk.read_le_f32()?,
+                                    elasticity_weight: chunk.read_le_f32()?,
+                                },
+                            )))
+                        } else {
+                            values.push(Variant::PhysicalProperties(PhysicalProperties::Default))
+                        }
+                    }
+
+                    for (value, referent) in values.into_iter().zip(&type_info.referents) {
+                        let instance = self.instances_by_ref.get_mut(referent).unwrap();
+                        instance.properties.push((canonical_name.clone(), value));
+                    }
```

Comment:
Because there is no interleaving or any interesting transforms involved, we should write this code using one loop, iterating through referents and deserializing values as we go. That lets us skip allocating a data structure that will be mostly unused (default physical properties) and simplifies this code.